ambry-network/src/main/java/com/github/ambry/network/http2/Http2NetworkClient.java

```diff
@@ -144,6 +174,12 @@ public void operationComplete(ChannelFuture future) throws Exception {
     if (requestsToDrop.size() != 0) {
       logger.warn("Number of requestsToDrop: {}", requestsToDrop.size());
       http2ClientMetrics.http2RequestsToDropCount.inc(requestsToDrop.size());
+      for (int correlationId : requestsToDrop) {
+        Channel streamChannel = correlationIdInFlightToChannelMap.remove(correlationId);
+        if (streamChannel != null) {
+          releaseAndCloseStreamChannel(streamChannel);
+        }
+      }
     }
 
     http2ClientResponseHandler.getResponseInfoQueue().poll(readyResponseInfos, pollTimeoutMs);
```

Comment:
Also, should probably remove the correlationIds for these responses from the inflight map.

Reply:
Good catch. Re-organized the map logic:
add: when a stream is acquired
remove: after `http2ClientResponseHandler.getResponseInfoQueue().poll()`

Reply:
Does this mean that a request cannot be dropped if stream acquisition takes a long time? I guess that is okay since that may be a rare occurrence and would require some complicated logic in the acquire listener to determine if the request can be sent or not.

Reply:
You are right, if a stream is not acquired, it's won't be in inflight map, so it can't be dropped. It's rare, we don't expect stream acquisition takes a long time. 

And since the inflight map's value is streamChannel(to close it in case of drop request), we can't add <correlationId, streamChannel> to inflight map before stream acquisition done. 

An extra time check can be made right after stream acquired if it's really need.